From 0beac519be8a846b1ed8040a7b076d4297f7595b Mon Sep 17 00:00:00 2001
From: lucy <fleshshell@cock.li>
Date: Sat, 9 Apr 2022 08:36:24 +0100
Subject: [PATCH] i have no idea what im doing


diff --git a/dwm.c b/dwm.c
index 5f16260..1dad7e9 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1055,6 +1055,14 @@ manage(Window w, XWindowAttributes *wa)
 	c->bw = borderpx;
 
 	wc.border_width = c->bw;
+
+        unsigned int Centerw;
+        unsigned int Centerh;
+        Centerw = c->mon->mx + (c->mon->mw - WIDTH(c));
+        Centerh = c->mon->my + (c->mon->mh - HEIGHT(c));
+        if (c->isfloating && c->h > (Centerh - 1) && c->w > (Centerw - 1)){
+        	wc.border_width = 0;
+        }
 	XConfigureWindow(dpy, w, CWBorderWidth, &wc);
 	XSetWindowBorder(dpy, w, scheme[SchemeNorm][ColBorder].pixel);
 	configure(c); /* propagates border_width, if size doesn't change */
@@ -1287,6 +1295,16 @@ resizeclient(Client *c, int x, int y, int w, int h)
 	c->oldw = c->w; c->w = wc.width = w;
 	c->oldh = c->h; c->h = wc.height = h;
 	wc.border_width = c->bw;
+	unsigned int Centerw;
+	unsigned int Centerh;
+	Centerw = c->mon->mx + (c->mon->mw - WIDTH(c));	
+	Centerh = c->mon->my + (c->mon->mh - HEIGHT(c));
+	if (c->isfloating && c->h > (Centerh - 1) && c->w > (Centerw - 1)){
+		wc.border_width = 0;
+	}
+	
+	
+	
 	XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
 	configure(c);
 	XSync(dpy, False);
